### PR TITLE
✨(UI, API, Tests): Created Parent Profile Feature

### PR DESCRIPTION
**Pull request by PxttytheThxtty**
- Created Parent Profile Page
- Created Edit Parent Profile Page
- Dynamically Populated both pages
- Enables parents to edit their profile information
- Unit Tests for the editing parent profile calls
- Redesigned the Parent Dashboard
- Dynamically populated the parents children
- Added the call and message functionality to the Parent Dashboard Page

### CODE REVIEW

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    margin-bottom: 2%;
+    color: #3C3C3C;
+    text-align: center;
+    font-family: TitanOne;
+    margin-top: 0px;
+}
+
+.parent-profile-div{
+    width: 90%;
+    // height: 90%;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:13]: commented out code (profile pic)

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+.profile-picture {
+    width: 80%;
+    height: 100%;
+}
+
+.parent-info-small{
+    height: 6%;
+    background-color: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:66]: var(--ion-background-color)

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    align-items: center;
+    color: #79BA9E;
+    margin-bottom: 12px;
+    box-shadow: 1px 2px 5px rgba(0, 0, 0, 0.13);
+    border-radius: 6px;
+}
+
+.parent-info-large{
+    height: 150px;
+    background-color: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:78]: var(--ion-background-color)

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+.parent-text2{
+    margin-left: 14px;
+    font-size: 100%;
+    width: 35%;
+    padding-top: 10px;
+}
+
+.parent-text-info{
+    background-color: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:103]: var(--ion-background-color)

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+    line-height: 170%;
+    margin-top: auto;
+    margin-bottom: auto;
+    height: 95%;
+    border: none;
+    border-radius: 5px;
+}
+
+.parent-text-info2{
+    background-color: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:116]: var(--ion-background-color)

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+    color: #3C3C3C;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:117]: var(--ion-color-dark)

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+    font-size: 100%;
+    line-height: 170%;
+    height: 20%;
+    margin-top: 5px;
+    border-radius: 5px;
+    width: 45%;
+    padding-left: 3px;
+    margin-left: 14px;
+    border-color: #3C3C3C;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:126]: var(--ion-color-dark)

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+.change-button{
+    padding: 10px 22px;
+    gap: 10px;
+    border: 2px solid #176560;
+    border-radius: 10px;
+    background-color: #69A88DB2;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:134]: not sure which colour was meant to be here

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+    width: 45%;
+    padding-left: 3px;
+    margin-left: 14px;
+    border-color: #3C3C3C;
+}
+
+.change-button{
+    padding: 10px 22px;
+    gap: 10px;
+    border: 2px solid #176560;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:132]: var(--ion-color-secondary)

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+        width: 100%;
+    }
+
+    .profile-details-div{
+        width: 75%;
+    }
+
+    .parent-text-info2{
+        width: 40%;
+        // padding-left: 6px;
```
Comment (ricecakesreece) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.scss:273]: commented out code

File: libs/parent-dashboard/feature/src/lib/parent-dashboard.ts [hunk 11]
```diff
@@ -1,8 +1,97 @@
...
+    address: "",
+    registered: false,
+    type: 0,
+    password: "",
+    number: "",
+  }
+
+  constructor(private serv: API)
+  {
+
```
Comment (ricecakesreece) [inline libs/parent-dashboard/feature/src/lib/parent-dashboard.ts:45]: format to single line

File: libs/parent-profile/feature/src/lib/parent-profile.component.scss [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+    margin-top: 0px;
+}
+
+.parent-profile-div{
+    width: 90%;
+    // height: 90%;
+    margin-top: 60px;
+    margin-left: auto;
+    margin-right: auto;
+    background-color: #69A88DB2;
```
Comment (ricecakesreece) [inline libs/parent-profile/feature/src/lib/parent-profile.component.scss:17]: var(--ion-color-primary)

File: libs/parent-profile/feature/src/lib/parent-profile.component.scss [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+.profile-picture-div{
+    width: 33.3%;
+    height: 100%;
+    float: left;
+}
+
+.details-div{
+    width: 40%;
+    height: 100%;
+    // float: left;
```
Comment (ricecakesreece) [inline libs/parent-profile/feature/src/lib/parent-profile.component.scss:31]: commented out and space

File: libs/parent-profile/feature/src/lib/parent-profile.component.scss [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+    height: 100%;
+}
+
+.parent-info-small{
+    height: 6%;
+    background-color: #F9F9F9;
+    display:flex;
+    font-family: Roboto;
+    align-items: center;
+    color: #79BA9E;
```
Comment (ricecakesreece) [inline libs/parent-profile/feature/src/lib/parent-profile.component.scss:69]: var(--ion-color-tertiary)

File: libs/parent-profile/feature/src/lib/parent-profile.component.scss [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+    margin-bottom: 12px;
+    box-shadow: 1px 2px 5px rgba(0, 0, 0, 0.13);
+    border-radius: 6px;
+}
+
+.parent-info-large{
+    height: 26%;
+    background-color: #F9F9F9;
```
Comment (ricecakesreece) [inline libs/parent-profile/feature/src/lib/parent-profile.component.scss:77]: var(--ion-background-color)

File: libs/parent-profile/feature/src/lib/parent-profile.component.scss [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+
+.parent-text{
+    margin-left: 12px;
+    font-size: 100%;
+    width: 35%;
+    margin: 10px;
+}
+
+.parent-text-info{
+    color: #3C3C3C;
```
Comment (ricecakesreece) [inline libs/parent-profile/feature/src/lib/parent-profile.component.scss:95]: var(--ion-color-dark)

File: libs/parent-profile/feature/src/lib/parent-profile.component.scss [hunk 17]
```diff
@@ -0,0 +1,238 @@
...
+    margin-top: auto;
+    margin-bottom: auto;
+}
+
+.change-button{
+    padding: 10px 22px;
+    gap: 10px;
+    border: 2px solid #176560;
+    border-radius: 10px;
+    background-color: #69A88DB2;
```
Comment (ricecakesreece) [inline libs/parent-profile/feature/src/lib/parent-profile.component.scss:117]: var(--ion-color-primary)

File: libs/parent-profile/feature/src/lib/parent-profile.component.ts [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+    name: "",
+    sname: "",
+    mID: "",
+    provider: "",
+  }
+
+
+  constructor(private serv: API)
+  {
+
```
Comment (ricecakesreece) [inline libs/parent-profile/feature/src/lib/parent-profile.component.ts:43]: format single line

File: libs/parent-profile/feature/src/lib/parent-profile.component.ts [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+          this.medAidDetails.plan = res.plan;
+          this.medAidDetails.name = res.name;
+          this.medAidDetails.sname = res.sname;
+          this.medAidDetails.mID = res.mID;
+          this.medAidDetails.provider = res.provider;
+        },
+        error=>{console.log("Error has occured with API: " + error);}
+      )
+  };
+
```
Comment (ricecakesreece) [inline libs/parent-profile/feature/src/lib/parent-profile.component.ts:89]: space

**Review by ricecakesreece — CHANGES_REQUESTED** [review]
Colours used to be converted to our standard ionic colours.

General formatting and some commented out scss code.

**Review by Ruben-Brits — APPROVED** [review]
Cool just address the comments then im happy

**Pull request merged**